contracts/LpSugar.vy

```diff
@@ -328,6 +422,19 @@ def byIndex(_index: uint256, _account: address) -> Lp:
   """
   pools: DynArray[address[3], MAX_POOLS] = self._pools()
 
+  pool: IPool = IPool(pools[_index][1])
+  factory: IPoolFactory = IPoolFactory(pools[_index][0])
+  token0: address = pool.token0()
+  token1: address = pool.token1()
+
+  is_cl_pool: bool = False
+  is_stable: bool = factory.getPool(token0, token1, 0) == pools[_index][1]
+
+  if not is_stable and factory.getPool(token0, token1, 1) != pools[_index][1]:
+    is_cl_pool = True
+
+  if is_cl_pool:
+    return self._byDataCL(pools[_index], _account)
```

Comment:
We should pass over the token addresses to the `_byData()` and `_byDataCL()`. Should save some gas.